src/pages/AppHub/AppHub.js

```diff
@@ -0,0 +1,155 @@
+import { useConfig, useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { PropTypes } from '@dhis2/prop-types'
+import { InputField, Pagination } from '@dhis2/ui'
+import { NoticeBox, CenteredContent, CircularLoader } from '@dhis2/ui'
+import React, { useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useDebounce } from 'use-debounce'
+import {
+    useQueryParams,
+    StringParam,
+    NumberParam,
+    withDefault,
+} from 'use-query-params'
+import { AppCard } from '../../components/AppCard/AppCard'
+import { AppCards as AppCards_ } from '../../components/AppCards/AppCards'
+import { getLatestVersion } from '../../get-latest-version'
+import styles from './AppHub.module.css'
+
+const query = {
+    appHub: {
+        resource: 'appHub/v2/apps',
+        params: params => ({
+            pageSize: 24,
+            ...params,
+        }),
+    },
+}
+
+const AppCards = ({ apps }) => {
+    const history = useHistory()
+    const getIconSrc = app => app.images.find(i => i.logo)?.imageUrl
+
+    return (
+        <AppCards_>
+            {apps.map(app => (
+                <AppCard
+                    key={app.id}
+                    iconSrc={getIconSrc(app)}
+                    appName={app.name}
+                    appDeveloper={
+                        app.developer.organisation || app.developer.name
+                    }
+                    appVersion={getLatestVersion(app.versions).version}
+                    onClick={() => history.push(`/app/${app.id}`)}
+                />
+            ))}
+        </AppCards_>
+    )
+}
+
+AppCards.propTypes = {
+    apps: PropTypes.array.isRequired,
+}
+
+const SearchFilter = ({ query, onQueryChange }) => (
+    <InputField
+        className={styles.searchField}
+        placeholder={i18n.t('Search AppHub apps')}
+        value={query}
+        onChange={({ value }) => onQueryChange(value)}
+        type="search"
+    />
+)
+
+SearchFilter.propTypes = {
+    onQueryChange: PropTypes.func.isRequired,
+    query: PropTypes.string,
+}
+
+const AppsList = ({ apps, pager, onPageChange }) => {
+    if (apps.length === 0) {
+        return (
+            <>
+                <h1 className={styles.header}>{i18n.t('No apps found')}</h1>
+                <p>No apps match your criteria</p>
```

Comment:
Any reason for this not being translated?

Would also be great to have a more "forgiving"/specialized message on the index-page (Preinstalled Core apps). Might not be worth the extra-props/complexity of reusing this component though, as Core apps should hopefully be on the App Hub before this is released.

Reply:
Nope - good catch :+1: 

The `AppsList` component already has props for custom messages/labels to use for available updates, headings and search input label so having an additional prop for the 'no apps match your criteria' message should be fine.